pandas/io/formats/printing.py

```diff
@@ -206,7 +206,7 @@ def as_escaped_unicode(thing, escape_chars=escape_chars):
 
         return compat.text_type(result)
 
-    if (compat.PY3 and hasattr(thing, '__next__')) or hasattr(thing, 'next'):
+    if hasattr(thing, '__next__') or hasattr(thing, 'next'):
```

Comment:
I think you can also remove the `or hasattr(thing, 'next')`: https://www.python.org/dev/peps/pep-3114/

Reply:
Done.